### DISCUSSION TITLE
v7.4.0

### DISCUSSION THREAD

**Discussion by Taiko2k** [Taiko2k/Tauon#825] [General]
## Overhauled Chromecast Support

Chromecast support in Tauon has received an overhaul. It now functions more like how it does in other apps. For instructions on how to use this, see wiki page [here](https://github.com/Taiko2k/TauonMusicBox/wiki/Chromecast). 

Specific changes:

- Now controlled by main UI controls. 
- Eliminated delay that was previously present.
- Now supports album art and metadata display on TV.
- Now supports volume control.
- Now supports remote Airsonic tracks.

> Note: Feature currently not enabled with Windows version

## Samplerate forwarding to DAC

 - You can now enable a setting "Avoid resampling", when enabled, Tauon will give the native samplerate of your media to the sound interface. Note that on Linux you'll need some additional configuration if you want this to reach your DAC. See [here](https://github.com/Taiko2k/TauonMusicBox/wiki/Audio-Specs) for details.

## Other changes

 - Multi disc albums now have a visual separator in tracklist
 - Fixed garbled audio on Raspberry Pi
 - Fixed cursor theme on Gnome Wayland
 - Fixed no sound bug with certain audio devices

<hr /><em>This discussion was created from the release <a href='https://github.com/Taiko2k/TauonMusicBox/releases/tag/'>v7.4.0</a>.</em>

**Comment by Solarunit** [comment]
Chromecast doesn't work for me. I see Chromecast mode is activated, but no sound. The only sound I hear is when adjusting volume in tauon.

____

Also Broadcast and Chromecast options in the menu is a bit confusing.

**Reply by Taiko2k** [reply]
What do you see happen on the TV?

**Reply by Solarunit** [reply]
It's not a TV, another Chromecast device without screen. I can only change its volume via tauon.

**Reply by Taiko2k** [reply]
And after starting chromecast mode and starting a track, the progress bar in tauon moves?

**Reply by Solarunit** [reply]
Nope

**Reply by Solarunit** [reply]
I tried it in the previous version, it's kinda worked with a huge delay, so it was unusable.

**Reply by Taiko2k** [reply]
If you run from Tauon from terminal, do you see an obvious error when trying?

**Reply by Solarunit** [reply]
will try

**Reply by Solarunit** [reply]
```
Scanning for chromecasts...
Start controller server
Start PHAzOR backend...
Using SDL version: 2.0.16
Applying theme: Cucumber Splash
Start PHAzOR backend...
```

Nothing in log when changing volume btw

**Reply by Solarunit** [reply]
also nothing when starting another track to play

**Reply by Taiko2k** [reply]
ok I give up

**Reply by Solarunit** [reply]
@Taiko2k I just tried to cast YouTube video from the Chromium flatpak and it worked without any delay!

**Reply by Solarunit** [reply]
in VLC ~~it doesn't work, even volume sound...~~
It's working now! After I disabled Firewall...

**Reply by Solarunit** [reply]
@Taiko2k maybe it's Firewall blocking tauon?

upd: yep! It's working now!
and no delay!

**Reply by Solarunit** [reply]
I think in the wiki page there should be some info about Firewall configuration for Tauon :thinking:

**Reply by Solarunit** [reply]
IIRC in the previous version Chromecast worked differently? You could cast song and at the same time listen to another song on a PC.

**Reply by Taiko2k** [reply]
Yeah I changed it to not play locally to be more in line with how Chromecasting is supposed to work

**Reply by Solarunit** [reply]
But it could be a nice feature to have.
